### PR TITLE
Upgrade postcss-values-parser to version 5

### PR DESCRIPTION
**Pull request by realityking**
Fixes #101

Version 5 is now using PostCSS 8 so there's no more necessary duplication holding back the upgrade.

This got more complex than expected, mostly because I was trying to use the upstream types. Unfortunately there are several errors in those types. I ended up copying (& fixing) the relevant parts into the definition file. When I get around to it, I'll submit a PR to the upstream to fix the types and the extra definition file can be eliminated.
As an alternative, I could simplify the definition file similar to the previous one and remove the type narrowing code. That would preclude using the upstream types in the future.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,7 +28,7 @@
     },
     "scripts": {
         "prepare": "tsc --outDir dist --declaration --declarationDir dist",
-        "test": "jest --ci",
+        "test": "jest",
```
Comment (joscha) [inline package.json]: can we keep the ci flag? https://jestjs.io/docs/cli#--ci

File: src/postcss-values-parser.d.ts [hunk 2]
```diff
@@ -1,26 +1,121 @@
 // incomplete declarations for the postcss-values-parser
 declare module 'postcss-values-parser' {
```
Comment (joscha) [inline src/postcss-values-parser.d.ts:2]: I see that v5 comes with it's own types now - are they still incomplete? Can we open a PR back to upstream with these changes here?

File: tsconfig.json [hunk 3]
```diff
@@ -1,7 +1,12 @@
 {
     "compilerOptions": {
         "module": "commonjs",
-        "target": "es2018"
+        "target": "es2018",
+        "paths": {
```
Comment (joscha) [inline tsconfig.json:5]: OOI, why this change?

**Comment by joscha** [comment]
> I'll submit a PR to the upstream to fix the types and the extra definition file can be eliminated.

Ah sorry, missed this when reading first. Link the PR here when you opened it!

File: src/postcss-values-parser.d.ts (hunk 2 above)
Comment (joscha) [reply src/postcss-values-parser.d.ts:2]: just saw your message on the PR description!

File: package.json (hunk 1 above)
Comment (realityking) [reply package.json]: That slipped in after my local testing, fixing it

**Comment by realityking** [comment]
There's 3 relevant PRs to this:

- https://github.com/postcss/postcss/pull/1549 Merged and released as postcss@8.2.9
- https://github.com/shellscape/postcss-values-parser/pull/132 Merged, not yet released
- https://github.com/shellscape/postcss-values-parser/pull/133 WIP

File: tsconfig.json (hunk 3 above)
Comment (realityking) [reply tsconfig.json:5]: This is necessary to make the override of types that exist in `postcss-values-parser` work. Once upstream is fixed, this can go away.

**Comment by joscha** [comment]
> There's 3 relevant PRs to this:

sorry for beings so slow - all of these have been merged, can this PR now simplified @realityking ?

**Comment by realityking** [comment]
While they’ve all been merged, shellscape/postcss-values-parser hasn’t yet made a release with the changes.

**Comment by joscha** [comment]
I've nudged on these pull requests :)

**Comment by joscha** [comment]
released in `detective-postcss@5.0.0`

**Comment by realityking** [comment]
❤️ 

I’ll send a clean-up PR once the upstream is updated
